common/str.cpp

```diff
@@ -444,6 +444,29 @@ uint String::hash() const {
 	return hashit(c_str());
 }
 
+void String::replace(uint pos, uint count, const String& str) {
```

Comment:
Same remark as in algorithm.h: the space goes before the & and not after.
And the same goes for pointers: the space goes before the *.
I am not going to comment again on that but obviously they should all be fixed ;)